aderyn_core/src/detect/low/state_variables_not_logged_in_event.rs

```diff
@@ -0,0 +1,229 @@
+use std::{
+    collections::{BTreeMap, HashMap},
+    error::Error,
+};
+
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    ast::{ASTNode, Expression, Mutability, NodeID, VariableDeclaration},
+    context::{
+        browser::{ExtractEmitStatements, GetImmediateChildren},
+        workspace_context::WorkspaceContext,
+    },
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use crate::{capture, context::browser::ExtractAssignments};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct StateVariableNotLoggedInEventDetector {
+    // All the state variables, set at the beginning of the detect Function
+    mutable_state_variables: HashMap<i64, VariableDeclaration>,
+    // Keys are: [0] source file name, [1] line number, [2] character location of node.
+    // Do not add items manually, use `capture!` to add nodes to this BTreeMap.
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for StateVariableNotLoggedInEventDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        /*
+         * How this detector works:
+         * Create a list of all state variables (this includes structs but not their subfields)
+         * Iterates through all functions
+         * Searches each function for assignments
+         * For each assignment found check if the left hand side of the assignment is a state variable using prior created list
+         * If LHS is a state variable then look at all emitted events in the same function
+         * Check if any of the parameters of found events match the state variable
+         * If not, raise an issue
+         */
+
+        //collect all mutable state variables together
+        self.mutable_state_variables = context
+            .variable_declarations()
+            .iter()
+            .filter_map(|&var_decl| {
+                if !var_decl.constant
+                    && matches!(var_decl.mutability(), Some(Mutability::Mutable))
+                    && var_decl.state_variable
+                {
+                    Some((var_decl.id, (*var_decl).clone())) // Deref and clone the VariableDeclaration.
```

Comment:
I don't think you need to deference here. Cloning can work on a ` &VariableDeclaration` -> ` VariableDeclaration`